### ISSUE TITLE
check that stim_channel exists in edf.py

### ISSUE THREAD

**Issue by nbara** [mne-tools/mne-python#4217]
This is a follow-up to an issue raised in https://github.com/mne-tools/mne-python/pull/4205.

In `read_raw_edf()`, it is possible to specify `stim_channel` as as `str`. However, if stim_channel doesn't exactly match one of ch_names (e.g. if you input 'status' instead of 'Status'), the code fails [line 517](https://github.com/mne-tools/mne-python/blob/master/mne/io/edf/edf.py#L518) because stim_channel is a `str`, instead of an index.

Technically stim_channel should get converted into an index at [line 463](https://github.com/mne-tools/mne-python/blob/master/mne/io/edf/edf.py#L463), but this never happens if the labels don't match exactly.

I suggest the following addition to `_get_edf_info()` to check that `stim_channel` is actually found in the data, and output a more helpful error message. 

```
from difflib import get_close_matches

if isinstance(stim_channel, str):
    if stim_channel not in ch_names:
        matches = get_close_matches(stim_channel, ch_names, n=1)[0]
        raise ValueError('Could not find a channel named "%s" in datafile'
                         ', closest match is "%s"' % (stim_channel, matches))
```

If you think this is a good idea, I can make a PR once #4205 is merged, or squeeze this into #4205.

**Comment by agramfort** [comment]
fine with me but I think that get_close_matches is a too loose check. I
would just check that strings match in lower/upper case eventually omitting
spaces.

**Comment by nbara** [comment]
Ok, so what about this then?

```
    if isinstance(stim_channel, str):
        if stim_channel not in ch_names:
            err = 'Could not find a channel named "%s" in datafile.' \
                  % stim_channel
            casematch = [ch for ch in ch_names
                         if stim_channel.lower().replace(' ', '') == \
                         ch.lower().replace(' ', '')]
            if casematch:
                    err += ' Closest match is "%s".' % casematch[0]
            raise ValueError(err)
```

Is it ok to include this in the open PR?

**Comment by agramfort** [comment]
ok

I would make a tiny private subfunction to do this though

**Comment by nbara** [comment]
Closed in favour of #4205
